hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hss/DispatchForResponseCodeHssCall.java

```diff
@@ -0,0 +1,132 @@
+
+    private final AccountID senderId;
+
+    @Nullable
+    private final TransactionBody syntheticBody;
+
+    private final VerificationStrategy verificationStrategy;
+    private final DispatchGasCalculator dispatchGasCalculator;
+    private final Set<Key> authorizingKeys;
+
+    /**
+     * Convenience overload that slightly eases construction for the most common case.
+     *
+     * @param attempt the attempt to translate to a dispatching
+     * @param syntheticBody the synthetic body to dispatch
+     * @param dispatchGasCalculator the dispatch gas calculator to use
+     */
+    public DispatchForResponseCodeHssCall(
+            @NonNull final HssCallAttempt attempt,
+            @Nullable final TransactionBody syntheticBody,
+            @NonNull final DispatchGasCalculator dispatchGasCalculator,
+            @NonNull final Set<Key> authorizingKeys) {
+        this(
+                attempt.enhancement(),
+                attempt.systemContractGasCalculator(),
+                attempt.addressIdConverter().convertSender(attempt.senderAddress()),
+                syntheticBody,
+                attempt.defaultVerificationStrategy(),
+                dispatchGasCalculator,
+                authorizingKeys);
+    }
+
+    /**
+     * More general constructor, for cases where perhaps a custom {@link VerificationStrategy} is needed.
+     *
+     * @param enhancement the enhancement to use
+     * @param senderId the id of the spender
+     * @param syntheticBody the synthetic body to dispatch
+     * @param verificationStrategy the verification strategy to use
+     * @param dispatchGasCalculator the dispatch gas calculator to use
+     */
+    // too many parameters
+    @SuppressWarnings("java:S107")
+    public DispatchForResponseCodeHssCall(
+            @NonNull final HederaWorldUpdater.Enhancement enhancement,
+            @NonNull final SystemContractGasCalculator gasCalculator,
+            @NonNull final AccountID senderId,
+            @Nullable final TransactionBody syntheticBody,
+            @NonNull final VerificationStrategy verificationStrategy,
+            @NonNull final DispatchGasCalculator dispatchGasCalculator,
+            @NonNull final Set<Key> authorizingKeys) {
+        super(gasCalculator, enhancement, false);
+        this.senderId = Objects.requireNonNull(senderId);
+        this.syntheticBody = syntheticBody;
+        this.verificationStrategy = Objects.requireNonNull(verificationStrategy);
+        this.dispatchGasCalculator = Objects.requireNonNull(dispatchGasCalculator);
+        this.authorizingKeys = authorizingKeys;
+    }
+
+    @Override
+    public @NonNull PricedResult execute(@NonNull final MessageFrame frame) {
+        if (syntheticBody == null) {
+            return gasOnly(
+                    haltResult(
+                            ERROR_DECODING_PRECOMPILE_INPUT,
+                            contractsConfigOf(frame).precompileHtsDefaultGasCost()),
```

Comment:
Should there be a config for `precompileHssDefaultGasGost`? (**Hss**)

Reply:
I don't think it is worth it to have a different gas cost for this for each system contract.

Reply:
Maybe should be renamed?